riscv/rocc.cc

```diff
@@ -2,11 +2,13 @@
 
 #include "rocc.h"
 #include "trap.h"
+#include "insn_macros.h"
 #include <cstdlib>
 
 #define customX(n) \
   static reg_t c##n(processor_t* p, insn_t insn, reg_t pc) \
   { \
+    require_accelerator; \
```

Comment:
This isn’t architecturally consistent. XS is not supposed to be a writable field: it’s a readable thing driven by the accelerator.

Reply:
One more thing I would like to confirm with you is how we decide whether a Z* extension is on or off.
First, they cannot be represented in the misa.
And if XS is read-only, that means that for any Z* extension, it should have at least a 1-bit status register to indicate whether the extension is on or off, in addition to implementing its own logical functions. 
And even if the extension is purely combinatorial, such as Zk extension, it should have a status register to indicate its state. It looks like the majority of extensions are currently focused on the functional specification and have not yet declared the address of their status registers.